TRAMS-API/Controllers/ProjectsController.cs

```diff
@@ -0,0 +1,102 @@
+    public class ProjectsController : Controller
+    {
+        private readonly IProjectsRepository _projectsRepository;
+        private readonly IAcademiesRepository _academiesRepository;
+        private readonly ITrustsRepository _trustsRepository;
+        private readonly IMapper<PostProjectsRequestModel, PostAcademyTransfersProjectsD365Model> _mapper;
+
+        public ProjectsController(IProjectsRepository projectsRepository,
+                                  IAcademiesRepository academiesRepository,
+                                  ITrustsRepository trustsRepository,
+                                  IMapper<PostProjectsRequestModel, PostAcademyTransfersProjectsD365Model> mapper)
+        {
+            _projectsRepository = projectsRepository;
+            _academiesRepository = academiesRepository;
+            _trustsRepository = trustsRepository;
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// API endpoint for creating a new Academy Transfers Project
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [Route("/projects/")]
+        [ProducesResponseType(typeof(string), StatusCodes.Status204NoContent)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status422UnprocessableEntity)]
+        public async Task<IActionResult> InsertTrust([FromBody]PostProjectsRequestModel model)
+        {
+            var debug = 0;
+
+            var projectAcademiesIds = model.ProjectAcademies.Select(a => a.AcademyId).ToList();
+
+            foreach(var academyId in projectAcademiesIds)
+            {
+                var academy = await _academiesRepository.GetAcademyById(academyId);
+
+                if (academy == null)
+                {
+                    return new UnprocessableEntityObjectResult($"No academy found with the id of: {academyId}");
+                }
+            }
+
+            var allTrustIds = new List<Guid>();
+
+            if (model.ProjectAcademies != null && model.ProjectAcademies.Any())
+            {
+                allTrustIds.AddRange(model.ProjectAcademies.Where(a => a.Trusts != null && a.Trusts.Any())
+                                                           .SelectMany(s => s.Trusts)
+                                                           .Select(s => s.TrustId));
+            }
+
+            if (model.ProjectTrusts != null && model.ProjectTrusts.Any())
+            {
+                allTrustIds.AddRange(model.ProjectTrusts.Select(p => p.TrustId));
+            }
+
+            if (allTrustIds.Any())
+            {
+                foreach (var trustId in allTrustIds.Distinct())
+                {
+                    var trust = _trustsRepository.GetTrustById(trustId);
+
+                    if (trust == null)
+                    {
+                        return new UnprocessableEntityObjectResult($"No trust found with the id of: {trustId}");
```

Comment:
Suggestion: As above, would it be worthwhile to return an aggregate result containing the full model state rather than just a single issue at a time?

Reply:
Will do